ql/src/java/org/apache/hadoop/hive/ql/parse/repl/metric/ReplicationMetricCollector.java

```diff
@@ -116,14 +136,16 @@ public void reportStageEnd(String stageName, Status status, long lastReplId,
         stage = new Stage(stageName, status, -1L);
       }
       stage.setStatus(status);
-      stage.setEndTime(System.currentTimeMillis());
+      stage.setEndTime(getCurrentTimeInMillis());
       stage.setReplSnapshotsCount(replSnapshotCount);
       if (replStatsTracker != null && !(replStatsTracker instanceof NoOpReplStatsTracker)) {
         String replStatString = replStatsTracker.toString();
         LOG.info("Replication Statistics are: {}", replStatString);
         stage.setReplStats(replStatString);
       }
       progress.addStage(stage);
+      // Check the progress string doesn't surpass the RM_PROGRESS column width.
+      checkRMProgressLimit(progress, stage);
```

Comment:
nit: 
it's not just check. it is updating the state as well. we should name it accordingly